Bump kube-rbac-proxy

oshoval opened the pull request:
<!-- Thanks for sending a pull request!

Before you click the 'Create pull request' make sure that:
- This PR introduces a single feature of fix, just one
- This PR does not leave the main branch broken
- Every commit in this PR has a commit message explaining what do you change,
  why and what is the outcome
- If your change introduces a complex concept or a change to user interaction
  with the project or the application, make sure to document it

If you don't comply with these rules, you waste your energy, time of reviewers
and cause suffering of future generations.
-->

**What this PR does / why we need it**:
Includes https://github.com/openshift/kube-rbac-proxy/pull/80
Affects CNAO and KMP.

`KubeRbacProxyImageDefault` was updated manually to a version
that contains the mention fix (we don't have automation, nor need atm for kube-rbac-proxy).

Before this PR we used `baedb268ac66456018fb30af395bb3d69af5fff3252ff5d549f0231b1ebb6901`
everywhere (via variable).

This PR did the following:
1. Changed from using variable to use static hash, because it is wrong to use variable which exists only at 
`pkg/components/components.go` but might be different between versions.
Since the image might change, it can't use the variable which exists only on `pkg/components/components.go`,
but should have an hard coded value per each release.
See https://github.com/kubevirt/cluster-network-addons-operator/pull/1563 for more info.

2. Now that [1] is fixed, manually bump the latest release candid (`test/releases/99.0.0.go`) and `components.go` to the new hash.

Note: since the tests validate the release on the upgrade test,
both for the old and latest, if there were any errors we would see.

**Special notes for your reviewer**:
We might want to backport it, but each release will need its own version (4.14, 4.13 etc).
Note that this affects only U/S, so not a must imo.

Automation is not needed but note that when a bump happens next time
only those file should be updated with the new hash: 
`pkg/components/components.go`
`test/releases/99.0.0.go`
It is safe because once one is updated, the tests would fail if the 2nd isn't.

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
4. If no release note is required, just write "NONE".
-->

```release-note
None
```

oshoval commented:
need to fix some stuff

RamLavi commented:
I don't think it's OK that we're retroactively changing the image on old versions. 
@oshoval can you elaborate why you want to do this?

oshoval commented:
We don't change it retroactively 
we just replace variable with the hash
see please first paragraph of 
https://github.com/kubevirt/cluster-network-addons-operator/pull/1563

RamLavi commented:
> We don't change it retroactively we just replace variable with the hash (each value reflects the one in its own test/release/X file) see please first paragraph of #1563

ah I get it..

oshoval commented:
> > We don't change it retroactively we just replace variable with the hash (each value reflects the one in its own test/release/X file) see please first paragraph of #1563
> 
> ah I get it.. The please add to the PR and commit desc this logic:
> 
> ```
> v0.0.11 and above (PR) --> registry.k8s.io/coredns/coredns@sha256:a0ead06651cf580044aeb0a0feba63591858fb2e43ade8c9dea45a6a89ae7e5e
> v0.0.10  --> registry.k8s.io/coredns/coredns@sha256:5b6ec0d6de9baaf3e92d0f66cd96a25b9edbce8716f5f15dcd1a616b3abd590e
> < v0.0.10  --> k8s.gcr.io/coredns/coredns@sha256:5b6ec0d6de9baaf3e92d0f66cd96a25b9edbce8716f5f15dcd1a616b3abd590e
> ```

It is not the same image

oshoval commented:
> > We don't change it retroactively we just replace variable with the hash (each value reflects the one in its own test/release/X file) see please first paragraph of #1563
> 
> ah I get it.. The please add to the PR and commit desc this logic:
> 
> ```
> v0.0.11 and above (PR) --> registry.k8s.io/coredns/coredns@sha256:a0ead06651cf580044aeb0a0feba63591858fb2e43ade8c9dea45a6a89ae7e5e
> v0.0.10  --> registry.k8s.io/coredns/coredns@sha256:5b6ec0d6de9baaf3e92d0f66cd96a25b9edbce8716f5f15dcd1a616b3abd590e
> < v0.0.10  --> k8s.gcr.io/coredns/coredns@sha256:5b6ec0d6de9baaf3e92d0f66cd96a25b9edbce8716f5f15dcd1a616b3abd590e
> ```

Took slightly different approach 
more details on PR desc (updated).
Also note that there are just 2 versions in game, so for all the old i changed to the old one
and for the latest i changed to the latest which was added only on this PR.

oshoval commented:
/cc @RamLavi 
can you please take another look ?

RamLavi reviewed (commented):
Can you please update the commit desc?

oshoval commented:
> Can you please update the commit desc?

It has the gist
```
KubeRbacProxyImageDefault was updated manually.
Each test/release/x.go file was updated according the version it had
in its variable.
Since the tests validate the release on the upgrade test,
both for the old and latest, if there were any errors we would see.
```
The rest isn't as important imho, if you find something specific you think should be added please elaborate

oshoval commented:
basically the gist was already there
added few more sentences

oshoval commented:
Added also a note about next time we change the hash in the PR desc

RamLavi commented:
> /cc @RamLavi can you please take another look ?

From what I see - you chose 2 URLs for different version. 
The commit desc should explain what is the logic behind it. 
This addresses it but I don't feel it's clear enough:
```
Each test/release/x.go file was updated according the version it had
in its variable (basically all stable to the previous hash, latest to the new hash).
```

Could you please elaborate?

oshoval commented:
Before this PR we used `baedb268ac66456018fb30af395bb3d69af5fff3252ff5d549f0231b1ebb6901`
everywhere

This PR did 2 things
1. Changed to use variable to static hash, because it is wrong to use variable which exists only at 
test/release/99.0.0.go but might be different between versions.
2. Now that [1] is fixed, manually bump on the latest and components.go to the new hash.

oshoval commented:
Updated PR desc with the above info

oshoval commented:
rebased

oshoval commented:
It seems some quay flaky issues, but didn't investigate deeper, and it is not just on this PR
/retest

oshoval commented:
/retest
